Item icons

Zsteven44 opened the pull request.

Zsteven44 committed:
feat: grid and linear button selectable, fixing item row ui

Zsteven44 committed:
feat: row item fixes and fields partially added

Zsteven44 committed:
feat: strings and for viewholder added, changes to row item xml

Zsteven44 committed:
feat: added picasso to gradle for icons

Zsteven44 committed:
feat: small changes

Zsteven44 committed:
feat: working on json deserialzing damage objects

Zsteven44 committed:
feat: row item fields correctly setting text precision

Zsteven44 committed:
feat: line decoration created

Zsteven44 committed:
fix: adjusted row item ui

Zsteven44 committed:
refactor: preliminary ui subject to change

Zsteven44 committed:
feat: removed tabviews

Zsteven44 committed:
fix: rebased to master

Zsteven44 committed:
feat: added imageurl field to item model

Zsteven44 committed:
feat: ammo icons added

Zsteven44 committed:
feat: added attachments imageurls to json

Zsteven44 committed:
fix: minor changes to json for attachments

Zsteven44 committed:
feat: added equipment icons to json

Zsteven44 committed:
feat: updated Consumables in Json with icons

Zsteven44 committed:
refactor: fixed bandage img url

Zsteven44 committed:
feat: vehicles added to json

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/decorations/ItemRowDecoration.java:
```diff
@@ -0,0 +1,46 @@
+import android.view.View;
+
+public class ItemRowDecoration extends RecyclerView.ItemDecoration{
+    private Drawable divider;
```
final

zafrani replied:
@NonNull

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/decorations/ItemRowDecoration.java:
```diff
@@ -0,0 +1,46 @@
+public class ItemRowDecoration extends RecyclerView.ItemDecoration{
+    private Drawable divider;
+
+    public ItemRowDecoration(Drawable divider) {
```
@Nonnull final

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/decorations/ItemRowDecoration.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Override
+    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
```
final all these

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/decorations/ItemRowDecoration.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Override
+    public void onDraw(Canvas canvas, RecyclerView parent, RecyclerView.State state) {
```
Final all of these

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/decorations/ItemRowDecoration.java:
```diff
@@ -0,0 +1,46 @@
+            RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child.getLayoutParams();
+
+            int dividerTop = child.getBottom() + params.bottomMargin;
+            int dividerBottom = dividerTop + divider.getIntrinsicHeight();
```
final these and the above variables

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+    private TextView stabilityTextView;
+    private TextView rateTextView;
+    private ImageView imageImageView;
+    private Context context;
```
final all of these variables

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+        if (itemAmmo != null) {
+            this.ammoTextView.setText(context.getString(R.string.row_item_ammo, itemAmmo));
+        } else {
+            this.ammoTextView.setText(" - ");
```
Make this value a static variable of the class

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+            this.ammoTextView.setText(" - ");
+        }
+    }
+    public void setItemMagazine(final Float itemMagazine) {
```
@Nullable

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+            this.magazineTextView.setText(" - ");
+        }
+    }
+    public void setItemCapacity(final Float itemCapacity) {
```
@Nullable

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+            this.damageTextView.setText(" - ");
+        }
+    }
+    public void setItemStability(final Float itemStability) {
```
@Nullable

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+            this.stabilityTextView.setText(" - ");
+        }
+    }
+    public void setItemRange(final Float itemRange) {
```
@Nullable

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+            this.rangeTextView.setText(" - ");
+        }
+    }
+    public void setItemRate(final Float itemRate) {
```
@Nullable

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,127 @@
+            this.rateTextView.setText(" - ");
+        }
+    }
+    public void setItemIcon(final String itemImage, String category) {
```
@Nullable

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/fragments/ItemTabFragment.java:
```diff
@@ -60,12 +62,15 @@ public void onViewCreated(@NonNull final View view,
         recyclerView = (RecyclerView) view.findViewById(R.id.fragment_itemtab_recyclerview);
         recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
-        final ItemListAdapter adapter = new ItemListAdapter(items);
+        Drawable dividerDrawable = getResources().getDrawable(R.drawable.item_row_divider_decoration);
```
final

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/fragments/ItemTabFragment.java:
```diff
@@ -60,12 +62,15 @@ public void onViewCreated(@NonNull final View view,
         recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
-        final ItemListAdapter adapter = new ItemListAdapter(items);
+        Drawable dividerDrawable = getResources().getDrawable(R.drawable.item_row_divider_decoration);
+        RecyclerView.ItemDecoration itemDividerDecoration = new ItemRowDecoration(dividerDrawable);
```
final

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/utils/FileUtil.java:
```diff
@@ -26,12 +28,20 @@ public static Items getItems(@NonNull final Activity activity) throws IOExceptio
+    public static Items bindItems(@NonNull Items items) {
+        List<Category> categories = items.getCategories();
+        for (Category category: categories){
+            category.updateChildren();
```
why is this needed?

Zsteven44 replied:
The constructor wasn't running updateChildren.

zafrani replied:
Make the parameter, list and each item final.

zafrani commented on app/src/main/res/drawable/item_row_divider_decoration.xml:
```diff
@@ -0,0 +1,6 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <size android:height="1dp" />
+    <solid android:color="#aaaaaa" />
```
Add this to colors.xml

Zsteven44 committed:
feat: made pr changes

Zsteven44 committed:
erge branch 'items-vehicles' into item-icons
i
i This is to merge pr with continued branch
 especially if it merges an updated upstream into a topic branch.

Zsteven44 committed:
fix: fileutil changes according to pr comments

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java:
```diff
@@ -35,6 +35,14 @@ public void onBindViewHolder(@NonNull final ItemViewHolder holder, @NonNull fina
+        holder.setItemRange(item.getRange());
+        holder.setItemRate(item.getRate());
+        holder.setItemStability(item.getStability());
+        holder.setItemIcon(item.getImageUrl(), item.getCategory());
```
why not use a holder.setItem function and set all of these from it?

Zsteven44 replied:
this changes in the next branch I pull request to a bind() function

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/decorations/ItemRowDecoration.java:
```diff
@@ -0,0 +1,53 @@
+
+        final int childCount = parent.getChildCount();
+        for (int i = 0; i < childCount - 1; i++) {
+            View child = parent.getChildAt(i);
```
final

Zsteven44 replied:
removed, replaced with view in xml.

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,128 @@
-    private TextView typeTextView;
-    private TextView categoryTextView;
+    private final String IMAGE_URL_BASE= "https://github.com/Zsteven44/PUBG-Companion-Images/blob/master/images/";
+    private final String ARG_RAW = "?raw=true";
```
this can be static

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ItemViewHolder.java:
```diff
@@ -1,35 +1,128 @@
-    private TextView nameTextView;
-    private TextView typeTextView;
-    private TextView categoryTextView;
+    private final String IMAGE_URL_BASE= "https://github.com/Zsteven44/PUBG-Companion-Images/blob/master/images/";
```
this can be static

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/models/Item.java:
```diff
@@ -17,13 +20,49 @@
+    @SerializedName("stability")
+    private final Float stability;
+    @SerializedName("rate")
+    private final Float rate;
```
These should be nullable.

zafrani commented on app/src/main/java/tech/zafrani/pubgapp/utils/FileUtil.java:
```diff
@@ -26,12 +28,20 @@ public static Items getItems(@NonNull final Activity activity) throws IOExceptio
         return null;
     }
 
+    public static Items bindItems(@NonNull final Items items) {
```
This should be non null

zafrani commented on app/src/main/res/drawable/item_row_divider_decoration.xml:
```diff
@@ -0,0 +1,6 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <size android:height="1dp" />
+    <solid android:color="@color/colorDivider" />
```
This is in the color file so it doesn't need to have color in its value. It can just be divider.

Zsteven44 replied:
removing this

zafrani commented on app/src/main/res/drawable/item_view_selector.xml:
```diff
@@ -0,0 +1,12 @@
+    <item android:background="@color/colorPrimary" android:state_selected="true">
+
+    </item>
+    <item android:background="@color/colorPrimaryDark" android:state_pressed="true">
```
Same here, this doesn't need color in the value.

Zsteven44 replied:
removed becuase its no longer being used.

Zsteven44 committed:
fix: fixes per pr comments

Zsteven44 committed:
fix: removed itemDecoration class
